migrations/Version20230407115407.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20230407115407 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Affect territoryof partner to all user without territory (except admin and usager)';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $users = $this->connection->fetchAllAssociative('SELECT id, email, partner_id, roles FROM user WHERE territory_id IS NULL AND roles NOT LIKE \'%"ROLE_USAGER"%\' AND roles NOT LIKE  \'%"ROLE_ADMIN"%\'');
```

Comment:
tu peux peut être tester directement dans la requête que partner_id IS NOT NULL non ?